core/src/Kokkos_View.hpp

```diff
@@ -400,6 +404,107 @@ class View : public Impl::BasicViewFromTraits<DataType, Properties...>::type {
   }
 #endif
 
+  // The following are shortcuts to allow implicit integer precision
+  // in offset calculations. If and when these are not needed, the
+  // data access operator should come from BaseView only.
+  KOKKOS_FUNCTION constexpr index_type compute_offset(
+      std::index_sequence<>) const {
+    return 0;
+  }
+
+  template <size_t Rank, class IndexOffset>
+  KOKKOS_FUNCTION constexpr index_type compute_offset(
+      std::index_sequence<Rank>, IndexOffset index_offset) const {
+    if constexpr (std::is_same_v<typename base_t::layout_type,
+                                 Kokkos::layout_stride>)
+      return index_offset * static_cast<IndexOffset>(m_map.stride(0));
+    else
+      return index_offset;
+  }
+
+  template <size_t... Ranks, class... IndexOffsets>
+    requires(
+        Kokkos::Impl::IsLayoutLeftPadded<typename base_t::layout_type>::value)
+  KOKKOS_FUNCTION constexpr index_type compute_offset(
+      std::index_sequence<Ranks...>, IndexOffsets... index_offsets) const {
+    using idx_type     = std::common_type_t<IndexOffsets...>;
+    idx_type indices[] = {static_cast<idx_type>(index_offsets)...};
+    // self-recursive fold trick from
+    // https://github.com/llvm/llvm-project/blob/96e1914aa2e6d8966acbfbe2f4d184201f1aa318/libcxx/include/mdspan/layout_left.h#L144
```

Comment:
Link in the comment is not correct
```suggestion
    // https://github.com/llvm/llvm-project/blob/96e1914aa2e6d8966acbfbe2f4d184201f1aa318/libcxx/include/__mdspan/layout_left.h#L144
```